From 375c2e9485682d6a7cc87c78f2ce808648d4b46f Mon Sep 17 00:00:00 2001
From: celadon <celadon@intel.com>
Date: Mon, 30 Jun 2025 14:48:09 +0000
Subject: [PATCH] [PATCH] CIV: Fix sensor mediation hal reporting

Add android-removed-permissions.xml to
remove sensor features not supported
on CML NUC.
Add mixins option to support disable
static sensor list reported by hal.

From change-Id - I7eacbf8b2ed646c3a70b2fe68a89fd7bc3be0bf4

Tracked-on: OAM-92946
Signed-off-by: Swee Yee Fonn <swee.yee.fonn@intel.com>
Signed-off-by: celadon <celadon@intel.com>

diff --git a/groups/device-specific/caas/android-removed-permissions.xml b/groups/device-specific/caas/android-removed-permissions.xml
new file mode 100644
index 00000000..18d3ef88
--- /dev/null
+++ b/groups/device-specific/caas/android-removed-permissions.xml
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (C) 2020 The Android Open Source Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+
+<permissions>
+    <!-- Disable support for hardware we don't support -->
+    <unavailable-feature name="android.hardware.sensor.compass" />
+    <unavailable-feature name="android.hardware.sensor.accelerometer" />
+    <unavailable-feature name="android.hardware.sensor.gyroscope" />
+    <unavailable-feature name="android.hardware.sensor.light" />
+    <unavailable-feature name="android.hardware.sensor.ambient_temperature" />
+</permissions>
diff --git a/groups/device-specific/caas/caas.mk b/groups/device-specific/caas/caas.mk
index c92aaea2..29df13fa 100644
--- a/groups/device-specific/caas/caas.mk
+++ b/groups/device-specific/caas/caas.mk
@@ -25,6 +25,9 @@ PRODUCT_FULL_TREBLE_OVERRIDE := false
 
 PRODUCT_DEFAULT_PROPERTY_OVERRIDES += ro.zygote=zygote64
 PRODUCT_COPY_FILES += system/core/rootdir/init.zygote64.rc:root/init.zygote64.rc
+PRODUCT_COPY_FILES += system/core/rootdir/init.zygote64_32.rc:root/init.zygote64_32.rc \
+    $(LOCAL_PATH)/android-removed-permissions.xml:system/etc/permissions/android-removed-permissions.xml
+
 
 BOARD_USE_64BIT_USERSPACE := true
 
diff --git a/groups/device-specific/caas/files.spec b/groups/device-specific/caas/files.spec
index ba23b3bf..679d9289 100644
--- a/groups/device-specific/caas/files.spec
+++ b/groups/device-specific/caas/files.spec
@@ -11,6 +11,7 @@ overlay: "configurations for SystemUI"
 system.prop: "system properties file"
 {{ref_target}}.mk: "product definition file"
 r2_{{target}}.mk: "Ring 2 target for P.car"
+android-removed-permissions.xml: "remove features not supported"
 start_flash_usb.sh: "script for flashing qcow2 in Qemu"
 start_android_qcow2.sh: "script for starting android in Qemu"
 setup_host.sh: "script for setting up virtulization environment on host"
diff --git a/groups/sensors/mediation/BoardConfig.mk b/groups/sensors/mediation/BoardConfig.mk
index 0adc8282..7ad4545d 100644
--- a/groups/sensors/mediation/BoardConfig.mk
+++ b/groups/sensors/mediation/BoardConfig.mk
@@ -1,5 +1,9 @@
 USE_SENSOR_MEDIATION_HAL := true
 
+{{#disable_static_sensor_list}}
+MEDIATION_HAL_DISABLE_STATIC_SENSOR_LIST := true
+{{/disable_static_sensor_list}}
+
 SOONG_CONFIG_NAMESPACES += senPlugin
 SOONG_CONFIG_senPlugin  += SENSOR_LIST
 SOONG_CONFIG_senPlugin_SENSOR_LIST := {{enable_sensor_list}}
diff --git a/groups/sensors/mediation/option.spec b/groups/sensors/mediation/option.spec
index ff58bddc..205bb9e4 100644
--- a/groups/sensors/mediation/option.spec
+++ b/groups/sensors/mediation/option.spec
@@ -1,2 +1,2 @@
 [defaults]
-enable_sensor_list = true
+enable_sensor_list = false
-- 
2.34.1

